src/NUnitFramework/framework/Internal/TestNameGenerator.cs

```diff
@@ -583,6 +588,42 @@ public override string GetText(MethodInfo method, object[] args)
             }
         }
 
+        private class ParamArgListFragment : NameFragment
+        {
+            private readonly int _maxStringLength;
+
+            public ParamArgListFragment(int maxStringLength)
+            {
+                _maxStringLength = maxStringLength;
+            }
+
+            public override string GetText(MethodInfo method, object[] args)
+            {
+                var sb = new StringBuilder();
+                args = args ?? new object[0];
+
+                var parameters = method.GetParameters();
+
+                sb.Append('(');
+                var count = Math.Max(args.Length, parameters.Length);
+
+                for (int i = 0; i < count; i++)
+                {
+                    if (i > 0) sb.Append(", ");
+
+                    var paramName = GetDisplayString(parameters.Length > i ? parameters[i].Name : "?", _maxStringLength).Trim('\"');
```

Comment:
GetDisplayString is intended for values and I don't think it's a good fit for parameter names.

Reply:
I would use a raw value with no length limit for the parameter name

Reply:
Ah, right. I'm okay with not applying the length limit, but if we do, we should extract the length-limiting behavior out of GetDisplayString and call that directly from here rather than going through GetDisplayString.